extract_wheels/lib/wheel.py

```diff
@@ -43,6 +61,15 @@ def dependencies(self, extras_requested: Optional[Set[str]] = None) -> Set[str]:
     def unzip(self, directory: str) -> None:
         with zipfile.ZipFile(self.path, "r") as whl:
             whl.extractall(directory)
+            # The following logic is borrowed from Pip:
+            # https://github.com/pypa/pip/blob/cc48c07b64f338ac5e347d90f6cb4efc22ed0d0b/src/pip/_internal/utils/unpacking.py#L240
+            for info in whl.infolist():
+                mode = info.external_attr >> 16
+                # if mode and regular file and any execute permissions for
+                # user/group/world?
+                if mode and stat.S_ISREG(mode) and mode & 0o111:
+                    name = os.path.join(directory, info.filename)
```

Comment:
The upstream code has an if/else guard to guarantee that the file is not a directory, i.e 
```python
if fn.endswith('/') or fn.endswith('\\')
```

I think this behaviour should be preserved.